tests/components/qwikswitch/test_init.py

```diff
@@ -98,18 +82,45 @@ async def test_sensor_device(hass, aioclient_mock):  # noqa: F811
             }
         }
     }
+    aioclient_mock.get("http://127.0.0.1:2020/&device", json=DEVICES)
+    listen_mock = aioclient_mock.get("http://127.0.0.1:2020/&listen", long_poll=True)
     await async_setup_component(hass, QWIKSWITCH, config)
-
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
     await hass.async_block_till_done()
+
     state_obj = hass.states.get("sensor.ss1")
     assert state_obj.state == "None"
 
+    listen_mock.queue_response(
+        json={"id": "@a00001", "name": "ss1", "type": "rel", "val": "4733800001a00000"},
+    )
+    await asyncio.sleep(0.01)
+    await hass.async_block_till_done()
+    state_obj = hass.states.get("sensor.ss1")
+    assert state_obj.state == "416"
+
+
+async def test_switch_device(hass, aioclient_mock):
+    """Test a switch device."""
+    config = {
+        "qwikswitch": {
+            "sensors": {"name": "ss1", "id": "@a00001", "channel": 1, "type": "imod"},
+            "switches": ["@000001"],
+        }
+    }
+    aioclient_mock.get("http://127.0.0.1:2020/&device", json=DEVICES)
+    listen_mock = aioclient_mock.get("http://127.0.0.1:2020/&listen", long_poll=True)
+    await async_setup_component(hass, QWIKSWITCH, config)
```

Comment:
There is a setup component contexthelper that should printing more info and helping to debug setup failures

Reply:
assert_setup_component In Ha.tests.common